### ISSUE TITLE
Xiaomi-Redmi-6

### ISSUE THREAD

**Issue by SwissChocolateBrownie** [phhusson/treble_experimentations#2535]
I have my hands on a Redmi 6 and it seems this device is fairly accepting of customization and so i have decided to try the treble gsi on it.
the device has the following as of now-
A -> A/B upgrade by ota update 
VNDK ver 28
Treble supported check

and i have decided to flash the treble gsi and i have chosen the arm32_binder64a/b variant for it and was wondering if there are somethings i shd be aware of before going further such as whether do i need a custom vendor or something device specific 

Any help is greatly appreciated thnq in advance!

**Comment by pcurz** [comment]
Make proper backup of everything before touching anything

**Comment by SwissChocolateBrownie** [comment]
> Make proper backup of everything before touching anything

Even better this device is quite literally empty lol

**Comment by SwissChocolateBrownie** [comment]
> Make proper backup of everything before touching anything

The device is completely empty and the question that I had was that since gsi is supposed to work using the seperation of the vendor implementation and Android,I shouldn't have to flash a custom vendor for this one right?

**Comment by SwissChocolateBrownie** [comment]
> Make proper backup of everything before touching anything
The device is empty and the question that was on my mind was that I don't need a custom vendor right?

**Comment by JonnyTech** [comment]
Correct, keep stock vendor, stock everything, just flash the GSI to the system partition..

**Comment by SwissChocolateBrownie** [comment]
Ok well i have unlocked my bootloader successfully and am now proceeding to install MIUI 11.0.4.0 since .5.0 dosent work apparently
I will update this once ive successfully tried it.

**Comment by SwissChocolateBrownie** [comment]
Well i installed the 12.1 version without gapps and it worked fine but the gapps version is now stuck on the setupwizard it goes upto accept googles services then re-enter pin and once i enter the pin it goes back to the google services accept screen and is like a loop

**Comment by SwissChocolateBrownie** [comment]
i managed to bypass that by clicking info when it says settings keeps stopping which opens settings, after which u can launch chrome install an updated version of android setup which wont break.

Everything seemed to have been working other than the camera which kept crashing and the flashlight was unusable aswell  and kept saying camera in use by other app,i tired enabling the HAL3 option in misc settings and also the SElinux permissive but to no avail
